include/readdy/kernel/singlecpu/actions/SCPUBreakBonds.h

```diff
@@ -0,0 +1,176 @@
+                                if (readdy::model::rnd::uniform_real() < 1 - std::exp(-rate * _timeStep)) {
+                                    recipe.removeEdge(edge);
+                                }
+                            }
+                        }
+                    }
+                    return std::move(recipe);
+                };
+                scalar rateDoesntMatter{1.};
+                readdy::model::top::reactions::StructuralTopologyReaction reaction("__internal_break_bonds",
+                                                                                   reactionFunction, rateDoesntMatter);
+
+                auto resultingTopologies = reaction.execute(*top, kernel);
+
+                auto &topologies = kernel->getSCPUKernelStateModel().topologies();
+                auto &model = kernel->getSCPUKernelStateModel();
+                const auto &context = kernel->context();
+
+                if (!resultingTopologies.empty()) {
+                    // we had a topology fission, so we need to actually remove the current topology from the
+                    // data structure
+                    topologies.erase(topologies.begin() + topologyIdx);
+                    //log::error("erased topology with index {}", event.topology_idx);
+                    assert(top->isDeactivated());
+
+                    for (auto &&newTopology : resultingTopologies) {
+                        if (!newTopology.isNormalParticle(*kernel)) {
+                            // we have a new topology here, update data accordingly.
+                            newTopology.updateReactionRates(
+                                    context.topologyRegistry().structuralReactionsOf(newTopology.type()));
+                            newTopology.configure();
+                            model.insert_topology(std::move(newTopology));
+                        } else {
+                            // if we have a single particle that is not of flavor topology, remove from topology structure!
+                            model.getParticleData()->entry_at(newTopology.getParticles().front()).topology_index = -1;
+                        }
+                    }
+
+                } else {
+                    if (top->isNormalParticle(*kernel)) {
+                        kernel->getSCPUKernelStateModel().getParticleData()->entry_at(
+                                top->getParticles().front()).topology_index = -1;
+                        topologies.erase(topologies.begin() + topologyIdx);
+                        //log::error("erased topology with index {}", event.topology_idx);
+                        assert(top->isDeactivated());
+                    }
+                }
+            }
+            ++topologyIdx;
+        }
+    }
+
+private:
+    SCPUKernel *kernel;
+
+    // Note that this also evaluates forces as a by-product, i.e. adds to the force property of particles
```

Comment:
This is because I use the already existing TopologyActions to calculate the energy term, whose `perform` method also evaluates the forces and adds them to the particle entries. Do you think  the TopologyActions should have another method to only calculate the energy? That would be the most obvious fix for me.